lib/rules/media-feature-name-value-no-unknown/index.js

```diff
@@ -0,0 +1,314 @@
+'use strict';
+
+const { TokenType, NumberType } = require('@csstools/css-tokenizer');
+const { isTokenNode, isFunctionNode } = require('@csstools/css-parser-algorithms');
+const {
+	isMediaFeature,
+	isMediaFeatureValue,
+	matchesRatioExactly,
+} = require('@csstools/media-query-list-parser');
+
+const atRuleParamIndex = require('../../utils/atRuleParamIndex');
+const parseMediaQuery = require('../../utils/parseMediaQuery');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const vendor = require('../../utils/vendor');
+const { lengthUnits } = require('../../reference/units');
+const { mathFunctions } = require('../../reference/functions');
+const {
+	mediaFeatureNameAllowedValueKeywords,
+	mediaFeatureNameAllowedValueTypes,
+	mediaFeatureNames,
+} = require('../../reference/mediaFeatures');
+
+const ruleName = 'media-feature-name-value-no-unknown';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (name, value) => `Unexpected value "${value}" for name "${name}"`,
+});
+
+const HAS_MIN_MAX_PREFIX = /^(?:min|max)-/i;
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/media-feature-name-value-no-unknown',
+};
+
+const resolutionUnits = new Set([
+	// Resolution
+	'dpi',
+	'dpcm',
+	'dppx',
+	// "x" is omitted in "reference/units".
```

Comment:
minor

we maybe want to add it there and create a separate set so that it can be imported here
i.e. it's probably a false negative for `unit-no-unknown`
not sure if it has to be done in this PR though

Reply:
Yes, I don't like that I had to add it here.

It might make more sense for rules that want to ignore `x` to create a copy of the `Set` and then call `unitsCopy.delete('x')`

That would make the exception local and explicit.

Other rules that do want to have the full and complete set of units would then be able to use `reference/units`.

Reply:
adding `resolutionUnits` to `reference/units` sounds good to me. 👍🏼 
Can you do it, please?